### PR TITLE
Improve logic to restart nodes when IP's are not generated

### PR DESCRIPTION
**Pull request by vavuthu**
Fixes: #4420 

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/csr.py [hunk 1]
```diff
@@ -187,10 +188,15 @@ def wait_for_all_nodes_csr_and_approve(timeout=900, sleep=10, expected_node_num=
...
+        # In vSphere deployment it sometime happens that VM doesn't get ip and
         # then we need to restart it to make our CI more stable and let the VM
-        # to get IP and continue with loading ignition config. The rester of
+        # to get IP and continue with loading ignition config. The restart of
         # the VMs happens only once in reboot_timeout (120 seconds).
         if vsphere_object and time.time() - start_time >= reboot_timeout:
             start_time = time.time()
-            vsphere_object.find_vms_without_ip_and_restart()
+            if not is_vms_without_ip:
+                vms_without_ip = vsphere_object.find_vms_without_ip()
```
Comment (petr-balogh) [inline ocs_ci/utility/csr.py:198]: with your change: find_vms_without_ip is taking name, datacenter_name, cluster_name as arguments but you don't pass any of those.

Or am I missing something?

Comment (vavuthu) [reply ocs_ci/utility/csr.py]: missed that. added params in latest commit

**Review by petr-balogh — APPROVED** [review]

**Comment by vavuthu** [comment]
> PR validation
> 
> Cluster Name:
> Cluster Configuration: conf/ocsci/fips.yaml conf/deployment/vsphere/upi_1az_rhcos_vsan_3m_6w.yaml
> PR Test Suite: deployment
> PR Test Path: tests/
> Additional Test Params:
> OCP VERSION: 4.7
> OCS VERSION: 4.7
> 
> [Job FAILED](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/879//testReport) (installation failed, tests not executed).

job failed in destroy job. Deploy job is successfull

```
11:42:18 - MainThread - ocs_ci.ocs.registry - INFO - Verified pvc is mounted on image-registry-8656d8684b-4jrg7 pod
11:42:18 - MainThread - ocs_ci.deployment.deployment - INFO - Done creating rook resources, waiting for HEALTH_OK
11:42:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
11:42:23 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
11:42:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-76dbc6f57f-wpxwx -- ceph health
11:42:39 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
11:42:39 - MainThread - ocs_ci.deployment.deployment - INFO - Patch thin storageclass as non-default
```
Hence marking veriifed

**Review by petr-balogh — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Comment by vavuthu** [comment]
verified : https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/3760/

**Pull request merged**